File: programming-concepts/good-programming-practice/visual-basic/identifiers.vb
' Raspberry Pi Foundation
' Developed as part of Ada Computer Science
' Usage licensed under CC BY-NC-SA 4.0
' Note: This file is designed to be copied out and compiled on your machine.
' To run this file you need to:
' 1. Copy the contents
' 2. Paste them into the Visual Basic IDE of your choice (Visual Studio, for example)
' 3. Compile the program
' 4. Run the program

Module Program

    Const GravityOnEarth As Double = 9.81
    Const GravityOnMars As Double = 3.711

    ' The Main subroutine is the default entry point for a VB program
    Sub Main()
        Dim weight As Double = 35.7
        Console.WriteLine(Conv(weight))
        Console.WriteLine(CalculateWeightOnMars(weight))
    End Sub

    ' First example
    Function Conv(ByVal a As Double) As Double
        Dim b As Double = a / 9.81 * 3.711
        Return b
    End Function

    ' Second example
    ' Convert a weight on Earth to the weight on Mars based on gravity
    Function CalculateWeightOnMars(ByVal weightOnEarth As Double) As Double
        Dim weightOnMars As Double = weightOnEarth / GravityOnEarth * GravityOnMars
        Return weightOnMars
    End Function

End Module

File: programming-concepts/good-programming-practice/c-sharp/identifiers.cs
/*
Raspberry Pi Foundation
Developed as part of Ada Computer Science 
Usage licensed under CC BY-NC-SA 4.0

Note: This file is designed to be copied out and compiled on your machine.
In order for it to compile properly you need to ensure that the project name is the same as the "namespace" in this file.
To run this file you need to:
1. Copy the contents
2. Paste them into the C# IDE of your choice (Visual Studio, for example)
3. Change the namespace to match your project (if necessary)
4. Compile the program
5. Run the program
*/

using System;

namespace AdaCodeSamples {
    class GoodProgrammingPractice {
        const double GravityOnEarth = 9.81;
        const double GravityOnMars = 3.711;

        // The Main method is the entry point for all C# programs
        public static void Main() {
            // Execute both subroutines and output the results
            double weight = 35.7;
            Console.WriteLine(Conv(weight));
            Console.WriteLine(CalculateWeightOnMars(weight));
        }


        // First example
        public static double Conv(double a) {
            double b = a / 9.81 * 3.711;
            return b;
        }


        // Second example
        // Convert a weight on Earth to the weight on Mars based on gravity
        public static double CalculateWeightOnMars(double weightOnEarth) {
            double weightOnMars = weightOnEarth / GravityOnEarth * GravityOnMars;
            return weightOnMars;
        }


    }
}